### PR TITLE
add-share-document-feature

### PR DESCRIPTION
**Pull request by DamianCzajkowski**

### CODE REVIEW

File: src/quokka_editor_back/routers/documents.py [hunk 1]
```diff
@@ -118,3 +99,17 @@ async def delete_document(
...
+
+@router.post("/share/{document_id}")
+async def share_document(
+    document_id: UUID,
+    current_user: Annotated[User, Depends(get_current_user)],
+):
+    await get_document(document_id=document_id, user=current_user)
+
+    token = create_access_token(
+        data={"document_id": str(document_id), "permissions": "edit"}
```
Comment (Michal-Suhak) [inline src/quokka_editor_back/routers/documents.py:112]: Should there always be edit permission? What with View permission?

File: src/quokka_editor_back/routers/documents.py [hunk 2]
```diff
@@ -49,50 +33,47 @@ async def get_document(document_id: UUID) -> Document:
...
 async def create_document(
-    document_payload: DocumentSchema,
     current_user: Annotated[User, Depends(get_current_user)],
 ):
     new_document = await Document.create(
-        title=document_payload.title,
-        content=json.dumps(document_payload.content).encode()
-        if document_payload.content
-        else None,  # TODO: remove None
+        title="Draft Document",
```
Comment (Michal-Suhak) [inline src/quokka_editor_back/routers/documents.py:46]: Probably missed const

File: src/quokka_editor_back/routers/documents.py [hunk 3]
```diff
@@ -49,50 +33,47 @@ async def get_document(document_id: UUID) -> Document:
...
-    )  # TODO: add pydantic model as a serializer and reduce response objects to
+@router.get("/")
+async def read_all(current_user: Annotated[User, Depends(get_current_user)]):
+    return await Document.all()
 
 
-@router.post("/")  # TODO: add response_model
+@router.post("/")
 async def create_document(
-    document_payload: DocumentSchema,
```
Comment (Michal-Suhak) [inline src/quokka_editor_back/routers/documents.py:61]: Why did you delete this?

File: src/quokka_editor_back/routers/documents.py (hunk 1 above)
Comment (DamianCzajkowski) [reply src/quokka_editor_back/routers/documents.py:112]: I forgot to add more permissions here.

File: src/quokka_editor_back/schema/user.py [hunk 4]
```diff
@@ -0,0 +1,9 @@
+from pydantic import BaseModel
+
+
+class UserSchema(BaseModel):
+    username: str
+    email: str | None = None
```
[the remark this answers is not in the document]
Comment (DamianCzajkowski) [reply src/quokka_editor_back/schema/user.py:6]: Yep. That was my idea. We can change it if you want

File: src/quokka_editor_back/routers/documents.py (hunk 2 above)
Comment (DamianCzajkowski) [reply src/quokka_editor_back/routers/documents.py:46]: no

File: src/quokka_editor_back/routers/documents.py (hunk 3 above)
Comment (DamianCzajkowski) [reply src/quokka_editor_back/routers/documents.py:61]: The idea behind this was to create document like in google docs style. Without providing anything. Just click on the desired template and create a document that is named draft

File: src/quokka_editor_back/schema/user.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (DamianCzajkowski) [reply src/quokka_editor_back/schema/user.py:6]: I will mark this in the code that we should refactor this part to remove completely username from db

File: Makefile [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+	docker compose run --rm --service-ports --no-deps api
+
+run-worker:
+	docker compose run --rm --service-ports --no-deps worker
+
+migrate:
+	docker compose run --rm --service-ports --no-deps api aerich upgrade
+
+add-migration:
+	docker compose run --rm --service-ports --no-deps api aerich migrate
```
Comment (Michal-Suhak) [inline Makefile:14]: up-bg will start the whole backend?

Comment (Michal-Suhak) [reply Makefile:14]: If not, can't you add, for example, `run-backend`?

Comment (DamianCzajkowski) [reply Makefile:14]: up-bg only runs background services. To run the whole backend just use docker compose up

**Review by Michal-Suhak — APPROVED** [review]
